src/Kiota.Builder/Refiners/ShellRefiner.cs

```diff
@@ -239,7 +239,9 @@ private static void AddParsableInheritanceForModelClasses(CodeElement currentEle
             new (x => x is CodeClass @class && @class.IsOfKind(CodeClassKind.RequestBuilder),
                 "System.CommandLine",  "Command", "RootCommand"),
             new (x => x is CodeClass @class && @class.IsOfKind(CodeClassKind.RequestBuilder),
-                "System.CommandLine.Invocation",  "CommandHandler"),
+                "Microsoft.Graph.Cli.Core.IO",  "OutputFormatterFactory"),
```

Comment:
Kiota generation results can only depend on abstractions (for compilation), as well as platform (.net) APIs. we cannot add a dependency to Microsoft.Graph.Cli.Core here. That factory might need to be moved.

Reply:
I forgot about that rule. If you have some time, I'd like to learn a little more about the challenge this would cause. Should I remove the `System.CommandLine` references as well?

Reply:
in theory yes, sorry I missed that on the initial PR, but then all of our generation story for the CLI goes down the drain. 
The more I think about it the more I think about this the more I believe we'll need to add a "special" kiota cli package, that only depends on abstractions but brings additional things we need to build, like the formatters, as well as additional packages like system.commandline.
Following the theory of Kiota, we should in fact have abstract definitions for all the system.commandline stuff, and than implement that in a core package, but I do feel like this would be a lot of additional work for not a ton of added value.
So I'm ok with having hard dependencies on some things in the generation so long as those dependencies are introduced by this intermediate package.
Thoughts?

Reply:
The rationale behind this requirement is to allow users to make decisions on what libraries to use in the generated code, right?

Reply:
correct